aion/lib/aion/room_channel/room.ex

```diff
@@ -73,15 +87,33 @@ defmodule Aion.RoomChannel.Room do
 
   @spec get_current_question(__MODULE__.t) :: Question.t
   def get_current_question(room) do
-    room.question
+    room.current_question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers((list Question.t), integer) :: new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    case questions do
+      [last_question] ->
+        remaining_questions = fetch_questions(room_id)
+        get_new_question_set(last_question, remaining_questions)
+
+      [next_question | remaining_questions] ->
+        get_new_question_set(next_question, remaining_questions)
+
+      [] ->
+        %{questions: [], current_question: nil, answers: []}
+    end
+  end
+
+  @spec get_new_question_set(Question.t, list Question.t) :: new_question_set
```

Comment:
I can see two functions here named `get_new_question_set` and `get_new_question_with_answers`. Seeing that I would expect that
 + the first one won't contain any answers (since you said it explicitly in the latter one)
 + the second one will probably return something else than `question_set` as you in the first one you said explicitly that you will get a question set and in the second one you only say about new question. Judging by the name I would expect this function to return sort of a data structure containing Question.t and Answer.t. 

I'm not criticizing any of the naming approaches, just be consequent.